File: src/main/java/de/com/fdm/bot/commands/UpdateCommand.java
package de.com.fdm.bot.commands;

import de.com.fdm.bot.Command;
import de.com.fdm.bot.Parameters;
import de.com.fdm.bot.api.WatchtowerService;
import de.com.fdm.twitch.tmi.TmiService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class UpdateCommand implements Command {
    private final WatchtowerService watchtowerService;
    private final TmiService tmiService;

    public UpdateCommand(
            @Autowired WatchtowerService watchtowerService,
            @Autowired TmiService tmiService
    ) {
        this.watchtowerService = watchtowerService;
        this.tmiService = tmiService;
    }

    @Override
    public String execute(Parameters params) {
        tmiService.send(params.getChannel(), "Updating...");

        watchtowerService.update();

        return "";
    }
}

File: src/main/java/de/com/fdm/bot/commands/UpdateCommand.kt
package de.com.fdm.bot.commands

import de.com.fdm.bot.Command
import de.com.fdm.bot.api.github.GithubService
import de.com.fdm.bot.api.watchtower.WatchtowerService
import de.com.fdm.bot.api.twitch.tmi.ReceiverMessage
import de.com.fdm.bot.api.twitch.tmi.TmiService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class UpdateCommand(
    @param:Autowired private val watchtowerService: WatchtowerService,
    @param:Autowired private val tmiService: TmiService,
    @param:Autowired private val githubService: GithubService
) : Command {
    override fun execute(receiverMessage: ReceiverMessage): String {
        val currentHash = System.getenv("COMMIT_SHA")
        val latestHash = githubService.getLatestCommit()

        if (currentHash == latestHash) {
            return "Already on latest commit!"
        }

        tmiService.send(receiverMessage.channel, "Updating...")
        watchtowerService.update()
        return ""
    }
}